posthog/temporal/batch_exports/bigquery_batch_export.py

```diff
@@ -63,6 +67,54 @@ async def create_table_in_bigquery(
     return table
 
 
+def get_bigquery_fields_from_record_schema(
+    record_schema: pa.Schema, known_json_columns: list[str]
+) -> list[bigquery.SchemaField]:
+    """Generate a list of supported BigQuery fields from PyArrow schema.
+
+    This function is used to map custom schemas to BigQuery-supported types. Some loss
+    of precision is expected.
+
+    Arguments:
+        record_schema: The schema of a PyArrow RecordBatch from which we'll attempt to
+            derive BigQuery-supported types.
+        known_json_columns: If a string type field is a known JSON column then use JSON
+            as its BigQuery type.
+    """
+    bq_schema: list[bigquery.SchemaField] = []
+
+    for name in record_schema.names:
+        pa_field = record_schema.field(name)
+
+        if pa.types.is_string(pa_field.type):
+            if pa_field.name in known_json_columns:
+                bq_type = "JSON"
+            else:
+                bq_type = "STRING"
+
+        elif pa.types.is_binary(pa_field.type):
+            bq_type = "BYTES"
+
+        elif pa.types.is_signed_integer(pa_field.type):
+            bq_type = "INT64"
```

Comment:
BigQuery only supports `INT64`, which appears to be signed. We could be optimistic and send any unsigned ints to an `INT64`, but I don't want to deal with that right now. We can see how this works out first and adjust accordingly.